File: backend-auth/src/main/java/cc/jktu/apps/auth/controller/AuthController.java
package cc.jktu.apps.auth.controller;

import cc.jktu.apps.auth.controller.req.LoginReq;
import cc.jktu.apps.auth.controller.req.UserAddOrUpdateReq;
import cc.jktu.apps.auth.service.UserService;
import cc.jktu.apps.common.CommonResp;
import cc.jktu.apps.common.dao.entity.UserEntity;
import cc.jktu.apps.common.exception.BadRequestException;
import cc.jktu.apps.common.util.BcryptUtil;
import cn.dev33.satoken.stp.StpUtil;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping
@Api(tags = "认证接口")
public class AuthController {

    private final UserService userService;

    @ApiOperation("登录")
    @PostMapping("/login")
    public CommonResp<Void> login(@RequestBody LoginReq req) {
        final String username = req.getUsername();
        final String password = req.getPassword();

        final UserEntity user = userService.getByUsername(username);
        final Boolean ret = BcryptUtil.checkPassword(password, user.getPassword());
        if (!ret) {
            throw new BadRequestException("密码错误");
        }
        StpUtil.login(user.getId());
        return CommonResp.emptyWithMsg("登录成功");
    }

    @ApiOperation("注册")
    @PostMapping("/register")
    public CommonResp<Void> addOne(@RequestBody UserAddOrUpdateReq req) {
        userService.addOne(req.getUsername(), req.getPassword());
        return CommonResp.emptyWithMsg("注册成功");
    }

}

File: backend-auth/src/main/kotlin/cc/jktu/apps/auth/controller/AuthController.kt
package cc.jktu.apps.auth.controller

import cc.jktu.apps.auth.controller.req.LoginReq
import cc.jktu.apps.auth.controller.req.UserAddOrUpdateReq
import cc.jktu.apps.auth.service.UserService
import cc.jktu.apps.common.exception.BadRequestException
import cc.jktu.apps.common.util.CommonResp
import cc.jktu.apps.common.util.checkPassword
import cc.jktu.apps.common.util.emptyRespWithMsg
import cc.jktu.apps.common.util.ok
import cn.dev33.satoken.stp.StpUtil
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping
@Tag(name = "认证接口")
class AuthController(
    private val userService: UserService
) {

    @Operation(summary = "登录")
    @PostMapping("/login")
    fun login(@RequestBody req: LoginReq): CommonResp<TokenInfo> {
        val username = req.username
        val password = req.password
        val user = userService.getByUsername(username)
        val ret = checkPassword(password, user.password ?: "")
        if (!ret) {
            throw BadRequestException("密码错误")
        }
        StpUtil.login(user.id)
        val tokenInfo = StpUtil.getTokenInfo()
        return TokenInfo(tokenInfo.tokenName, tokenInfo.tokenValue).ok()
    }

    @Operation(summary = "注册")
    @PostMapping("/register")
    fun addOne(@RequestBody req: UserAddOrUpdateReq): CommonResp<Nothing?> {
        userService.createUser(req.username, req.password)
        return emptyRespWithMsg("注册成功")
    }
}

data class TokenInfo(val tokenName: String, val tokenValue: String)